### PR TITLE
Split UPE - Update names and add icons to the block and classic checkouts

### LINKED ISSUE #2843: [Split PE - Design] Update payment method names and add icons
Internal P2: pc4etw-LY-p2

With https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2827 and https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2820 merged into our `add/deferred-intent` base branch, we have now implemented the fundamentals for dPE (Split PE with deferred intents).

This issue is to make sure our implementation matches the proposed designs.

### Acceptance criteria
- [ ] Update the names of our Payment Methods (remove the "Pay with " from the name)
- [ ] Include the payment method icon

| Current | Designs |
|:-------:|:--------:|
| image | image |

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #2843

## Changes proposed in this Pull Request:

This PR:

- Updates the old icons in the assets folder.
- Adds a new icon for the combined card icon. I took this one from the existing one that Mayisha used on the settings page. 
- Updated all the payment method titles to remove the "Pay with" - this is inline with the designs as outlined on the issue (#2843). 
- Added the icons to the front end block and classic checkout pages. 

| Block checkout | Classic checkout |
|--------|--------|
| Screenshot 2024-01-29 at 12 05 14 pm | Screenshot 2024-01-29 at 10 07 03 am |

## Testing instructions

**Euros**

1. Change the store currency to Euros
2. On the Stripe plugin settings page:
   - add Stripe credentials that have all Euro payment methods enabled.
   - Enable all available APMs
4. Create a classic/shortcode checkout and a block checkout page.
1. Add a product to the cart view the Block and shortcode checkout pages. 
2. Confirm all the card and payment methods are displayed with icons etc.

**Brazil (Boleto)**

1. Change the store currency to Brazilian Real
2. On the Stripe plugin settings page:
   - add Stripe credentials that have all Brazilian payment methods enabled.
   - Enable all available APMs
4. Create a classic/shortcode checkout and a block checkout page.
1. Add a product to the cart view the Block and shortcode checkout pages. 
2. Confirm the card and Boleto are displayed with icons etc.

| Block checkout | Classic checkout |
|--------|--------|
| Screenshot 2024-01-29 at 12 01 34 pm| Screenshot 2024-01-29 at 12 04 37 pm | 

**Mexico (Oxxo)**

1. Change the store currency to Mexican Peso
3. On the Stripe plugin settings page:
   - add Stripe credentials that have all Mexico payment methods enabled.
   - Enable all available APMs
4. Create a classic/shortcode checkout and a block checkout page.
1. Add a product to the cart view the Block and shortcode checkout pages. 
2. Confirm the card and Oxxo methods are displayed with icons etc.

| Block checkout | Classic checkout |
|--------|--------|
| Screenshot 2024-01-29 at 12 00 18 pm| Screenshot 2024-01-29 at 11 59 46 am| 

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: includes/payment-methods/class-wc-stripe-upe-payment-method-giropay.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ class WC_Stripe_UPE_Payment_Method_Giropay extends WC_Stripe_UPE_Payment_Method
 	public function __construct() {
 		parent::__construct();
 		$this->stripe_id            = self::STRIPE_ID;
-		$this->title                = __( 'Pay with giropay', 'woocommerce-gateway-stripe' );
+		$this->title                = __( 'giropay', 'woocommerce-gateway-stripe' );
```
Comment (Mayisha) [inline includes/payment-methods/class-wc-stripe-upe-payment-method-giropay.php:21]:
```suggestion
		$this->title                = __( 'Giropay', 'woocommerce-gateway-stripe' );
```

**Review by Mayisha — APPROVED** [review]
All the scenarios look good. The icons are rendered beside the payment method name :+1:
- [x] Euro 
- [x] Brazil (Boleto)
- [x] Mexico (Oxxo)

File: includes/payment-methods/class-wc-stripe-upe-payment-method-giropay.php (hunk 1 above)
Comment (james-allan) [reply includes/payment-methods/class-wc-stripe-upe-payment-method-giropay.php:21]: Believe it or not, that's the correct capitalisation for giropay 🙈. [WooPayments uses the same all lowercase form](https://github.com/Automattic/woocommerce-payments/blob/edf69a59e2a8f0913c55ba27bf69bd1a1d551d2b/includes/payment-methods/class-giropay-payment-method.php#L27). [Stripe in their documentation](https://stripe.com/docs/payments/giropay) also refer to it like this.

Comment (Mayisha) [reply includes/payment-methods/class-wc-stripe-upe-payment-method-giropay.php:21]: My bad :smile:

**Pull request merged**